Userland/Libraries/LibWebView/CMakeLists.txt

```diff
@@ -25,8 +28,32 @@ set(GENERATED_SOURCES
     ../../Services/RequestServer/RequestServerEndpoint.h
     ../../Services/WebContent/WebContentClientEndpoint.h
     ../../Services/WebContent/WebContentServerEndpoint.h
+    ../../Services/WebContent/WebDriverClientEndpoint.h
+    ../../Services/WebContent/WebDriverServerEndpoint.h
+    ../../Services/WebSocket/WebSocketClientEndpoint.h
+    ../../Services/WebSocket/WebSocketServerEndpoint.h
     NativeStyleSheetSource.cpp
 )
 
 serenity_lib(LibWebView webview)
-target_link_libraries(LibWebView PRIVATE LibCore LibFileSystemAccessClient LibGfx LibGUI LibIPC LibProtocol LibJS LibWeb)
+target_link_libraries(LibWebView PRIVATE LibCore LibGfx LibGUI LibIPC LibProtocol LibJS LibWeb)
+
+if (SERENITYOS)
+    target_link_libraries(LibWebView PRIVATE LibFileSystemAccessClient)
+endif()
+
+if (NOT SERENITYOS)
+    foreach(header ${GENERATED_SOURCES})
+        get_filename_component(extension ${header} EXT)
+        if (NOT "${extension}" STREQUAL ".h")
+            continue()
+        endif()
+
+        get_filename_component(subdirectory ${header} DIRECTORY)
+        string(REGEX REPLACE "^\\.\\./\\.\\./" "" subdirectory "${subdirectory}")
+
+        install(FILES "${CMAKE_CURRENT_BINARY_DIR}/${header}" DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/${subdirectory}")
+    endforeach()
+
+    install(FILES "${SERENITY_PROJECT_ROOT}/Userland/Services/RequestServer/ConnectionCache.h" DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/RequestServer")
```

Comment:
Yeah tbh I have no idea what the lagom-install folder is actually for. But I just moved this from Lagom/CMakeLists.txt because Lagom just doing `add_subdirectory(LibWebView)` does not let it see any variables defined here (I tried exposing these as `LIBWEBVIEW_GENERATED_SOURCES` and leaving the install rules in Lagom alone).

Reply:
Oh hmm, so these are generated by some Services CMakeLists.txt, which Lagom doesn't see at all. So the closest client library for these generated headers *is* LibWebView on Lagom - e.g. for WebContent and RequestServer, there's no LibWebContent or LibRequestServer to hook into

Reply:
Eh. I guess this is fine. Not sure how to organize it atm to not be a bit hank